homeassistant/components/aladdin_connect/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for Aladdin Connect cover integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aladdin_connect import AladdinConnectClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    acc = AladdinConnectClient(data["username"], data["password"])
+    try:
+        login = await hass.async_add_executor_job(acc.login)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        raise CannotConnect from ex
+    else:
+        if not login:
+            raise InvalidAuth
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aladdin Connect."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
```

Comment:
These are missing in strings.json

Reply:
This is interesting as I ran one of the scripts that is supposed to generate and clean the translation up. I actually thought it moved this over to the en.json file at the time, but now don't see it :(   Can't remember where I read about the script, obviously it was a bad choice (or I simply executed it incorrectly).

Reply:
The translation script fixes the translations but the string.json needs to be done manually first.